### ISSUE TITLE
Both 'added' and 'removed' query triggered when adding component/removing entity in same frame

### ISSUE THREAD

**Issue by nickschot** [MozillaReality/ecsy#121]
I'm creating/removing an entity and on that entity a component based on a button click. Clicking repeatedly on this button adds/removes the entity as wanted. 

When clicking rapidly it might happen that, for a system listening to the component, both an added and a removed query are triggered in the same world execution/frame.

This causes the awkward situation where the "added" query is run without the component being actually present anymore (it is marked as a removed component).

Minimal reproduction here: https://jsfiddle.net/cdLheys3/1/

**Comment by nickschot** [comment]
I hope the issue is clear like

**Comment by fernandojsg** [comment]
Hi @nickschot thanks for reporting this. 
We have been discussing about how to handle this cases as most ideas involve a performance hit. eg: storing the commands each time you add or remove a component. Additionally we could include a `hasAddedComponent` and `getAddedComponent` similar to the `removed*` version of them.
But by now I have implemented an additional parameter on the `hasComponent` and `getComponent` that will let you query for all the components it doesn't matter if they are removed or alive.
Basically you can add a second parameter with `true` to query all the components.
```javascript
entity.getComponent(Component, true);
entity.hasComponent(Component, true);
```

That should work for your specific use case, while we will discuss on another issue a generic solution.

fyi is already landed on dev if you want to try it, in the incoming days I'll push a new release on master
